src/main/java/com/amazonaws/services/kinesis/clientlibrary/proxies/KinesisProxy.java

```diff
@@ -220,7 +220,8 @@ public KinesisProxy(final KinesisClientLibConfiguration config, final AmazonKine
                 config.getListShardsBackoffTimeInMillis(),
                 config.getMaxListShardsRetryAttempts());
         this.credentialsProvider = config.getKinesisCredentialsProvider();
-        this.streamArn = config.getStreamArn();
+        Arn arn = config.getStreamArn();
+        this.streamArn = arn != null ? arn.toString() : null;
```

Comment:
Would you kindly provide an in-line comment on why this is happening, and/or provide rationale as Javadoc on the member variable? We shouldn't force future maintainers to reason about intent -- especially since the effect here could be mistaken as [primitive obsession](https://wiki.c2.com/?PrimitiveObsession).